workers/handshake-manager/src/manager/matching/internal_engine.rs

```diff
@@ -133,47 +92,47 @@ impl HandshakeExecutor {
                     }
                 },
             }
+
+            // If matching failed, remove the other order from the candidate set
+            match_candidates.remove(&other_order_id);
         }
 
-        info!("No internal matches found for {order:?}");
         Ok(())
     }
 
+    /// Get the set of match candidates for an order
+    ///
+    /// Shuffles the ordering of the other orders for fairness
```

Comment:
Have been thinking about this - do we want to shuffle? I think it's worth considering ordering either by earliest -> latest creation date, or by size, to ensure we match up the biggest fills

Reply:
Also, there's no explicit shuffle going on here, we just insert the orders into a hash set

Reply:
yeah the shuffle is unnecessary, we can order them however we like down the line